packages/library-base/src/commonMain/kotlin/io/realm/kotlin/types/MutableRealmInt.kt

```diff
@@ -0,0 +1,227 @@
+ * correctly. For instance, if the value of a `counter` field for the object representing a `User`
+ * named Fred is currently 0, then the following code, executed on two different devices,
+ * simultaneously, even if connected by a slow, unreliable network, will **always** cause the value
+ * of `counter` to eventually converge on the value 2.
+ * ```
+ * realm.write {
+ *      query<User>("name = $0", "Fred")
+ *          .first()
+ *          .find()!!
+ *          .counter
+ *          .increment(1)
+ * }
+ * ```
+ * Note that the [set] operator must be used with care. It will quash the effects of any prior calls
+ * to [increment] or [decrement]. Although the value of a `MutableRealmInt` will always converge
+ * across devices, the specific value on which it converges will depend on the actual order in which
+ * operations took place. Mixing [set] with [increment] and/or [decrement] is, therefore, not
+ * advised, unless fuzzy counting is acceptable.
+ *
+ * `MutableRealmInt`s **cannot be primary keys**.
+ *
+ * `MutableRealmInt`s cannot store `null` values. However, it is possible to declare nullable
+ * `MutableRealmInt` class members:
+ * ```
+ * var counter: MutableRealmInteger? = null
+ * ```
+ *
+ * A reference to a managed `MutableRealmInt` is subject to all of the constraints that apply to the
+ * model object from which it was obtained: It can only be mutated within a transaction and it
+ * becomes invalid if the Realm backing it is closed. Note that a reference to a managed
+ * `MutableRealmInt` retains a reference to the model object to which it belongs. For example in
+ * this code:
+ * ```
+ * val counter: MutableRealmInt = realm.query<User>("user = $0", "Fred")
+ *      .first()
+ *      .find()!!
+ *      .counter
+ * ```
+ * the `counter` holds a reference to the `User` model object from which it was obtained. Neither
+ * can be GCed until all references to both are unreachable.
+ */
+public abstract class MutableRealmInt : Comparable<MutableRealmInt>, Number() {
+
+    /**
+     * Gets the `MutableRealmInt` value.
+     *
+     * @return the value.
+     */
+    public abstract fun get(): Long
+
+    /**
+     * Sets the `MutableRealmInt` value.
+     *
+     * Calling [set] forcibly sets the `MutableRealmInt` to the provided value. Doing this removes
+     * the effects of any calls to [increment] and [decrement] received before the call to `set`.
+     *
+     * @param value new value.
+     */
+    public abstract fun set(value: Number)
+
+    /**
+     * Increments the `MutableRealmInt`, adding the value of the argument. Increment/decrement
+     * operations from all devices are reflected in the new value, which is guaranteed to converge.
+     * This operation should not be confused with the [inc] operator, which returns a new value
+     * whereas this one modifies the value itself.
+     *
+     * @param value quantity to be added to the `MutableRealmInt`.
+     */
+    public abstract fun increment(value: Number)
```

Comment:
Do we really need these (also `decrement`), when there is other ways to achieve the same with the operators? Just adds more test cases, but really convenience 🤷

Reply:
Whao, that feels really weird to me ... wonder what the reasoning about `.inc` and `.dec` not changing the value is 🤔